opentelemetry-aws/src/trace/xray_propagator.rs

```diff
@@ -144,7 +145,8 @@ pub fn span_context_from_str(value: &str) -> Option<SpanContext> {
             ))
         }
         Err(trace_state_err) => {
-            otel_error!(name: "SpanContextFromStr", error = format!("{:?}", TraceError::Other(Box::new(trace_state_err))));
+            let error = TraceError::Other(Box::new(trace_state_err));
+            otel_error!(name: "SpanContextFromStr", error = format!("{:?}", error));
```

Comment:
I don't think this whole wrapping and boxing is needed to display it as a string
```suggestion
            otel_error!(name: "SpanContextFromStr", error = format!("{:?}", trace_state_err))
```

Reply:
you are right, this works